python/neuroglancer/pipeline/storage.py

```diff
@@ -38,59 +44,76 @@ def __init__(self, layer_path='', n_threads=20):
         self._n_threads = n_threads
 
         if self._path.protocol == 'file':
-            self._interface = FileInterface
+            self._interface_cls = FileInterface
         elif self._path.protocol == 'gs':
-            self._interface = GoogleCloudStorageInterface
+            self._interface_cls = GoogleCloudStorageInterface
         elif self._path.protocol == 's3':
-            self._interface = S3Interface
+            self._interface_cls = S3Interface
+
+        self._interface = self._interface_cls(self._path)
+
+        self._queue = Queue.Queue(maxsize=0) # infinite size
```

Comment:
Imagine I have a large affinity file, of 10gb. And I have only 8gb on my machine.
I write a for loop to to get chunks from the file and to upload them. 
The will all get quickly push to the queue and the program will crash.

The only solution will be to do storage.wait() every few put_file. 
The second solution is to limit the size of the queue.

Both solutions do the same under the hood. The only difference is that the first requires a deeper understanding of the storage class by the user, which is what one try to avoid when encapsulating code.